lib/build.sh

```diff
@@ -58,14 +58,21 @@ install_or_reuse_node_modules() {
   local build_dir=$1
   local layer_dir=$2
   local local_lock_checksum
+  local cached_lock_checksum
+
+  if ! detect_package_lock "$build_dir"; then
+    install_modules "$build_dir" "$layer_dir"
+    return 0
+  fi
+
+  touch "$layer_dir.toml"
 
   local_lock_checksum=$(sha256sum "$build_dir/package-lock.json" | cut -d " " -f 1)
+  cached_lock_checksum=$(yj -t < "$layer_dir.toml" | jq -r ".metadata.package_lock_checksum")
```

Comment:
what happens if `$layer_dir.toml` doesn't exist?

Also might be good to use `${layer_dir}.toml`. Even though `.` is not allowed as a variable char, I've seen weirder things happen.

Reply:
[Line 68](https://github.com/heroku/nodejs-npm-buildpack/pull/9/files#diff-cbea48db73996bd0b8adcff9b6b7fbd2R68) will create the file.. if it doesn't exist there's an error in the build when parsing the file, so I wanted to prevent that whether this layer has been created or not